words-master/scripts/script.js

```diff
@@ -1,113 +1,147 @@
-      letterStyle.transition = "";
-      if (currLetter === secretLetter) {
-        letterStyle.backgroundColor = "green";
-      } else if (secretWord.indexOf(currLetter) !== -1) {
-        letterStyle.backgroundColor = "#ffff00";
-      } else {
-        letterStyle.backgroundColor = "#C0C0C0";
+      for (let i = currIdx; i < currSize; i++) {
+        const letterStyle = letters[i].style;
+        const currWordLetter = currWord[idx];
+        const secretWordLetter = secretWord[idx++];
+
+        const numOfLetCurrWord = numOfLetter(arrCurrWord, currWordLetter);
+        const numOfLetSecretWord = numOfLetter(arrSecretWord, currWordLetter);
+
+        const chopIndex = arrCurrWord.indexOf(currWordLetter);
+
+        letterStyle.transition = "0.1s";
+
+        if (currWordLetter === secretWordLetter) {
+          letterStyle.backgroundColor = "green";
+          if (numOfLetCurrWord > numOfLetSecretWord) {
+            arrCurrWord.splice(chopIndex, 1);
+          }
+        } else if (
+          !numOfLetSecretWord ||
+          numOfLetCurrWord > numOfLetSecretWord
+        ) {
+          if (numOfLetCurrWord > numOfLetSecretWord) {
+            arrCurrWord.splice(chopIndex, 1);
+          }
+          letterStyle.backgroundColor = "#C0C0C0";
+        } else if (numOfLetSecretWord) {
+          letterStyle.backgroundColor = "#ffff00";
+        }
       }
-      idx++;
+      if (currWord === secretWord) console.log("You winner!!");
+      this.currWord = "";
+      this.currIdx = index;
     }
-    if (currWord === secretWord) console.log("You winner!!");
-    currWord = "";
-    currIdx = index;
   }
-}
 
-function isLetter(letter) {
-  if ((letter >= "a" && letter <= "z") || (letter >= "A" && letter <= "Z")) {
-    if (letter.length === 1 || letter === "Backspace") return true;
+  isLetter(letter) {
+    if ((letter >= "a" && letter <= "z") || (letter >= "A" && letter <= "Z")) {
+      if (letter.length === 1) return true;
+    }
+    return false;
   }
-  return false;
-}
 
-restartBtn.addEventListener("click", () => {
-  getSekretWord();
-  currWord = "";
-  index = 0;
-  currIdx = 0;
-  valid = false;
-  for (const letter of letters) {
-    letter.innerText = "";
-    letter.style.backgroundColor = "";
+  numOfLetter(word, letter) {
+    let count = 0;
+    for (const ell of word) {
+      if (ell === letter) count++;
+    }
+    return count;
   }
-});
+  restart() {
+    let { getSekretWord, currWord, index, currIdx, valid } = game;
+    getSekretWord();
+    currWord = "";
+    index = 0;
+    currIdx = 0;
+    valid = false;
+    for (const letter of letters) {
+      letter.innerText = "";
+      letter.style.backgroundColor = "";
+    }
+  }
+}
+
+const game = new GameMetods();
+
+{
+  if (!game.secretWord.length) game.getSekretWord();
+}
 
-body.addEventListener("keydown", addLeter);
+restartBtn.addEventListener("click", game.restart);
+body.addEventListener("keydown", game.addLetter);
```

Comment:
try to use a lambda wrapper to avoid `this` problems.
```js
restartBtn.addEventListener('click', () => game.restart());
body.addEventListener('keydown', (event) => game.addLetter(event));
```